python/ngen_conf/src/ngen/config/multi.py

```diff
@@ -22,7 +23,7 @@ class MultiBMI(BMIParams, smart_union=True):
     # NOTE this is derived from the list of modules
     main_output_variable: Optional[str]
     #NOTE aliases don't propagate to subclasses, so we have to repeat the alias
-    model_name: Optional[str] = Field(alias="model_type_name")
+    model_name: str = Field(alias="model_type_name")
```

Comment:
Could this still be optional?

Reply:
I don't think so. In the super class it is a `str`. I think the most sensible thing is to either set a default empty string or leave it as is. Thoughts?

Reply:
well the validator `build_model_name` was using the optional `None` in order to build the name from the named nested modules.  We can set a default empty string `""` but would need to ensure that validator recognizes that and substitutes the built name.

Reply:
Yeah, this is one of my gripes with `pydantic`. There is not a semantic way to signify to the caller that something is computed if it is not provided. The way that we have been doing this in DMOD is to set the default of a non-optional field to `None`. I guess in a perfect world, we would create a constant, say `COMPUTED` and use that as the default value.

Here, I just went with setting the default value to `None` and updated the root validator to always set the model name to something if it is not provided.